src/awst_build/eb/arc4/base.ts

```diff
@@ -85,3 +101,103 @@ class Arc4EqualsFunctionBuilder extends FunctionBuilder {
     )
   }
 }
+
+export class Arc4EncodedFromBytesFunctionBuilder extends FunctionBuilder {
+  constructor(
+    sourceLocation: SourceLocation,
+    private ptypeFactory?: (args: PType[]) => ARC4EncodedType,
+    private genericArgsCount?: number,
+  ) {
+    super(sourceLocation)
+  }
+  call(args: ReadonlyArray<NodeBuilder>, typeArgs: ReadonlyArray<PType>, sourceLocation: SourceLocation): NodeBuilder {
+    const funcName = 'fromBytes'
+    const {
+      ptypes,
+      args: [initialValueBuilder],
+    } = parseFunctionArgs({
+      args,
+      typeArgs,
+      genericTypeArgs: this.genericArgsCount ?? 1,
+      funcName,
+      argSpec: (a) => [a.required(bytesPType)],
+      callLocation: sourceLocation,
+    })
+
+    const ptype = (this.ptypeFactory ? this.ptypeFactory(ptypes) : ptypes[0]) as ARC4EncodedType
+    const initialValue = initialValueBuilder.resolveToPType(bytesPType).resolve()
+    const initialValueExpr =
```

Comment:
nodeFactory.reinterpretCast actually handles this optimization for constant byte values internally

Reply:
It has been updated in this commit, 6c31fad.